shell/shell.go

```diff
@@ -57,3 +58,37 @@ func RmLink(last, path string) (string, error) {
 func AddLink(last, path, hash string) (string, error) {
 	return Exec("object", "patch", "add-link", last, path, hash)
 }
+
+func keyExists(key string) bool {
+	keys, err := Exec("key", "list", "-l")
+	if err != nil {
+		return false
+	}
+	lines := strings.Split(keys, "\n")
+	for _, line := range lines {
```

Comment:
can you do `range strings.Split(keys, "\n")` for better readability? same for `keyNames`

Reply:
Okay

Reply:
Done